pybossa/api/project.py

```diff
@@ -73,11 +73,15 @@ def _preprocess_post_data(self, data):
             data["info"]["data_access"] = ["L4"]
 
     def _create_instance_from_request(self, data):
-        if 'password' not in data.keys():
+        # password required if not syncing
+        sync = "sync" in data["info"]
+        if 'password' not in data.keys() and not sync:
             raise BadRequest("password required")
-        password = data.pop("password")
+        password = data.pop("password", "")
```

Comment:
This could result in setting up blank password. Hence lines 78-80 can be modified as below to confirm 
1. password cannot be blank when call is not from proj sync
2. with proj sync, proj password hash to be part of payload
```python
password = data.pop("password", "")
if not (sync or password) or (sync and 'passwd_hash' not in data):
    raise BadRequest("password required")
```

Reply:
Right now it will return a res 400 Bad Request which will show up as: https://github.com/bloomberg/pybossa/blob/master/pybossa/view/projects.py#L3090

There is already a special case for a 415 that provides a more user friendly error, we could do something similar for 400s? Most of the changes I made to the API result in 400 bad requests if something isn't right.

Reply:
Meant to include, this is the 415 special case: https://github.com/bloomberg/pybossa/blob/master/pybossa/view/projects.py#L3082

Reply:
Thanks for the details. Looks like there's good logging in place to troubleshoot generic errors and error reporting to user for specific case. With above code, password hash during sync won't be an issue anymore. lgtm.